File: scalanativelib/api/src/ScalaNativeWorkerApi.scala
//package mill.scalanativelib.api
//
//import upickle.default.{macroRW, ReadWriter => RW}
//import java.io.File
//import sbt.testing.Framework
//
//trait ScalaNativeWorkerApi {
//  def discoverClang: os.Path
//  def discoverClangPP: os.Path
//  def discoverTarget(clang: os.Path, workDir: os.Path): String
//  def discoverCompileOptions: Seq[String]
//  def discoverLinkingOptions: Seq[String]
//
//  def config(nativeLibJar: os.Path,
//             mainClass: String,
//             classpath: Seq[os.Path],
//             nativeWorkdir: os.Path,
//             nativeClang: os.Path,
//             nativeClangPP: os.Path,
//             nativeTarget: String,
//             nativeCompileOptions: Seq[String],
//             nativeLinkingOptions: Seq[String],
//             nativeGC: String,
//             nativeLinkStubs: Boolean,
//             releaseMode: ReleaseMode,
//             logLevel: NativeLogLevel): NativeConfig
//
//  def defaultGarbageCollector: String
//  def nativeLink(nativeConfig: NativeConfig, outPath: os.Path): os.Path
//
//  def newScalaNativeFrameWork(framework: Framework, id: Int, testBinary: File,
//                              logLevel: NativeLogLevel, envVars: Map[String, String]): Framework
//}
//

File: scalanativelib/api/src/ScalaNativeWorkerApi.java
package mill.scalanativelib.api;
import sbt.testing.Framework;

public interface ScalaNativeWorkerApi {
    java.io.File discoverClang();
    java.io.File discoverClangPP();
    String discoverTarget(java.io.File clang, java.io.File workDir);
    String[] discoverCompileOptions();
    String[] discoverLinkingOptions();

    NativeConfig config(java.io.File nativeLibJar,
                        String mainClass,
                        java.io.File[] classpath,
                        java.io.File nativeWorkdir,
                        java.io.File nativeClang,
                        java.io.File nativeClangPP,
                        String nativeTarget,
                        String[] nativeCompileOptions,
                        String[] nativeLinkingOptions,
                        String nativeGC,
                        boolean nativeLinkStubs,
                        ReleaseMode releaseMode,
                        NativeLogLevel logLevel);

    String defaultGarbageCollector();
    java.io.File nativeLink(NativeConfig nativeConfig, java.io.File outPath);

    Framework newScalaNativeFrameWork(Framework framework, int id,
                                      java.io.File testBinary,
                                      NativeLogLevel logLevel,
                                      java.util.Map<String, String> envVars);
    }
